testlib/src/main/kotlin/io/spine/protodata/testing/PipelineSetup.kt

```diff
@@ -0,0 +1,276 @@
+            outputDir: Path,
+            settingsDir: Path,
+            writeSettings: (SettingsDirectory) -> Unit
+        ): PipelineSetup {
+            val callingClass = detectCallingClass()
+            val classLoader = callingClass.classLoader
+            val inputRoot = inputRootOf(language, classLoader)
+            val request = loadRequest(classLoader)
+            return PipelineSetup(
+                plugins,
+                inputRoot,
+                outputDir,
+                request,
+                settingsDir,
+                writeSettings,
+            )
+        }
+
+        /**
+         * Creates an instance assuming that the input directory and [CodeGeneratorRequest] are
+         * placed into the resources using [ProtoTap](https://github.com/SpineEventEngine/ProtoTap).
+         *
+         * The pipeline to be created will handle code generation in [Java].
+         *
+         * @param plugins
+         *         the list of plugins to be passed to the created pipeline.
+         * @param outputDir
+         *         the root directory to which the updated code will be placed.
+         * @param settingsDir
+         *         the directory to which store the settings for the given plugin.
+         * @param writeSettings
+         *         a callback for writing plugin settings before the pipeline is created.
+         */
+        public fun byResources(
+            plugins: List<Plugin>,
+            outputDir: Path,
+            settingsDir: Path,
+            writeSettings: (SettingsDirectory) -> Unit
+        ): PipelineSetup = byResources(Java, plugins, outputDir, settingsDir, writeSettings)
+
+        @VisibleForTesting
+        internal fun detectCallingClass(): Class<*> {
+            val thisClass = this::class.java
+            val caller = findCallerOf(thisClass, 0)
+            check(caller != null) { "Unable to obtain the caller for the class `$thisClass`." }
+            val callingClass = Class.forName(caller.className)
+            return callingClass
+        }
+
+        private fun loadRequest(classLoader: ClassLoader): CodeGeneratorRequest {
+            val file = Resource.file(
+                "$PROTOC_PLUGIN_NAME/$CODE_GENERATOR_REQUEST_FILE",
+                classLoader
+            )
+            file.open().use {
+                val request = CodeGeneratorRequest.parseFrom(it)
+                return request
+            }
+        }
+
+        private fun inputRootOf(language: Language, classLoader: ClassLoader): Path {
+            val languageDir = language.protocOutputDir()
+            val dirName = "$PROTOC_PLUGIN_NAME/$languageDir"
+            val dir = ResourceDirectory.get(dirName, classLoader)
+            val inputRoot = dir.toPath()
+            return inputRoot
+        }
+    }
+}
+
+@VisibleForTesting
+internal fun Language.protocOutputDir(): String {
+    return when(this) {
+        Java, Kotlin -> name.lowercase()
+        TypeScript -> "ts"
```

Comment:
This is in fact configurable. Since TypeScript is not a built-in, but rather a number of custom plugins, each of them will be able to set its own folder name.

So I would mention this in the docs, as a convention we introduce.

Reply:
Described in a section in the class doc.